src/runtimes/node/index.ts

```diff
@@ -62,6 +65,12 @@ const zipFunction: ZipFunction = async function ({
 
   const inSourceConfig = await findISCDeclarationsInPath(mainFile, name, featureFlags)
   const runtimeAPIVersion = inSourceConfig.runtimeAPIVersion === 2 ? 2 : 1
+
+  if (runtimeAPIVersion === 2) {
```

Comment:
Rather than overriding the value of feature flags if we detect the V2 API, I think we should take the value of `runtimeAPIVersion` into account when deciding the module format to use.

I think this was already in place for NFT (https://github.com/netlify/zip-it-and-ship-it/blob/main/src/runtimes/node/bundlers/nft/es_modules.ts#L80-L84), but I believe you changed it to use esbuild instead, so we lost that?

Reply:
Good idea! done in https://github.com/netlify/zip-it-and-ship-it/pull/1489/commits/a0ad3a3d90613bb7cdf27ac5c449e70c8345a89e